thoth/package_extract/core.py

```diff
@@ -47,20 +48,31 @@ def extract_buildlog(input_text: str) -> typing.List[dict]:
 def extract_image(image_name: str, timeout: int = None, *, registry_credentials: str = None,
                   tls_verify: bool=True) -> dict:
     """Extract dependencies from an image."""
-    image_name = quote(image_name)
-    with tempfile.TemporaryDirectory() as dir_path:
-        download_image(
-            image_name,
-            dir_path,
-            timeout=timeout or None,
-            registry_credentials=registry_credentials or None,
-            tls_verify=tls_verify
-        )
+    """Begin the timer for when the job starts"""
+    prometheus_registry = CollectorRegistry()
+    _metric_analyzer_job = Gauge('package_extract_time','Runtime of package extract job', registry=prometheus_registry)
+    with _metric_analyzer_job.time():
+            image_name = quote(image_name)
+            with tempfile.TemporaryDirectory() as dir_path:
```

Comment:
For better readability and lower complexity, this can be reduced to:
```python
with metric_analyzer_job.time(), tempfile.TemporaryDirectory() as dir_path:
        image_name = quote(image_name)
        ...
```